vpr/test/test_edge_groups.cpp

```diff
@@ -0,0 +1,31 @@
+#include <array>
+#include <utility>
+#include <cstddef>
+#include <set>
+
+#include "catch.hpp"
+
+#include "rr_graph.h"
+
+namespace {
+
+TEST_CASE("edge_groups_create_sets", "[vpr]") {
+    // Start with 4 unconnected edges: 1-2, 3-4, 5-6, 7-8
+    // Then connect connect odd/even edges with another edge
+    // Finally connect the odd and even edges with another edge
+    // Also add another unconnected edge 9-10
+    std::array<std::pair<int, int>, 9> edges{{{1, 2}, {3, 4}, {5, 6}, {7, 8}, {2, 5}, {4, 7}, {6, 7}, {9, 10}}};
```

Comment:
Probably worth testing this test case with some of the edge pairs reversed?

Reply:
I replaced the test with one that runs 200 tests with random node IDs and edge order, sort of like the `random_shuffle` node reordering algorithm.